src/platform/CatalogProvider.js

```diff
@@ -0,0 +1,80 @@
+// @flow
+
+import React, { useContext, useEffect, useMemo, useState } from "react";
+
+import type { AppManifest, AppBranch, AppPlatform } from "./types";
+import { isSupported, matchPlatform, matchBranches } from "./logic";
+
+import api from "./api";
+
+type State = {
+  apps: AppManifest[],
+};
+
+type Props = {
+  children: React$Node,
+};
+
+const initialState = {
+  apps: [],
+};
+
+export const PlatformCatalogContext = React.createContext<State>(initialState);
+
+const PlatformCatalogProvider = ({ children }: Props) => {
+  const [state, setState] = useState(initialState);
+
+  useEffect(() => {
+    api.fetchManifest().then((manifest) => setState({ apps: manifest }));
+  }, []);
+
+  return (
+    <PlatformCatalogContext.Provider value={state}>
+      {children}
+    </PlatformCatalogContext.Provider>
+  );
+};
+
+export const useCatalog = (
```

Comment:
We need a way to "get manifests from all branches" at once or else we will lose ordering. Product insisted they want to fine tune ordering at the beginning (before using more advanced ordering later).